src/main/webapp/app/file-upload-assessment/file-upload-assessment.component.ts

```diff
@@ -0,0 +1,482 @@
+                    // Set maximum width
+                    interact.modifiers!.restrictSize({
+                        min: { width: this.resizableMinWidth, height: 0 },
+                        max: { width: this.resizableMaxWidth, height: 2000 },
+                    }),
+                ],
+                inertia: true,
+            })
+            .on('resizestart', function(event: any) {
+                event.target.classList.add('card-resizable');
+            })
+            .on('resizeend', function(event: any) {
+                event.target.classList.remove('card-resizable');
+            })
+            .on('resizemove', function(event) {
+                const target = event.target;
+                // Update element width
+                target.style.width = event.rect.width + 'px';
+                target.style.minWidth = event.rect.width + 'px';
+            });
+
+        this.interactResizableTop = interact('.resizable-horizontal')
+            .resizable({
+                // Enable resize from bottom edge; triggered by class .resizing-bar-bottom
+                edges: { left: false, right: false, top: false, bottom: '.resizing-bar-bottom' },
+                // Set min height
+                modifiers: [
+                    interact.modifiers!.restrictSize({
+                        min: { width: 0, height: this.resizableMinHeight },
+                    }),
+                ],
+                inertia: true,
+            })
+            .on('resizestart', function(event: any) {
+                event.target.classList.add('card-resizable');
+            })
+            .on('resizeend', function(event: any) {
+                event.target.classList.remove('card-resizable');
+            })
+            .on('resizemove', function(event) {
+                const target = event.target;
+                // Update element height
+                target.style.minHeight = event.rect.height + 'px';
+                $('#submission-area').css('min-height', event.rect.height - 100 + 'px');
+            });
+    }
+
+    public ngOnDestroy(): void {
+        this.changeDetectorRef.detach();
+    }
+
+    public addReferencedFeedback(): void {
+        const referencedFeedback = new Feedback();
+        referencedFeedback.credits = 0;
+        this.referencedFeedback.push(referencedFeedback);
+        this.validateAssessment();
+    }
+
+    public deleteAssessment(assessmentToDelete: Feedback): void {
+        const indexToDelete = this.referencedFeedback.indexOf(assessmentToDelete);
+        this.referencedFeedback.splice(indexToDelete, 1);
+        this.validateAssessment();
+    }
+
+    /**
+     * Load next assessment in the same page.
+     * It calls the api to load the new unassessed submission in the same page.
+     * For the new submission to appear on the same page, the url has to be reloaded.
+     */
+    assessNextOptimal() {
+        if (this.exercise.type === ExerciseType.FILE_UPLOAD) {
```

Comment:
Why do we need to check this here? Isn't the component exclusively used for file upload exercises?

Reply:
Yes, I removed the check